### PR TITLE
[nn] lstm : no batch dim support

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #60585

TODO:
* [x] Update docs

### CODE REVIEW

File: torch/nn/modules/rnn.py [hunk 1]
```diff
@@ -725,6 +739,19 @@ def forward(self, input, hx=None):  # noqa: F811
                                   dtype=input.dtype, device=input.device)
             hx = (h_zeros, c_zeros)
         else:
+            if batch_sizes is None:  # If not PackSequence input.
```
Comment (kshitij12345) [inline torch/nn/modules/rnn.py:742]: These checks are here to make JIT happy.

File: test/test_modules.py [hunk 2]
```diff
@@ -363,8 +363,11 @@ def _can_be_noncontiguous(obj):
                     grad_output = default_output.clone().detach_().normal_()
                     default_output.backward(grad_output, retain_graph=True)
                 else:
-                    grad_output = tuple(o.clone().detach_().normal_() for o in default_output)
-                    for o, g_o in zip(default_output, grad_output):
+                    grad_output = tuple(self._traverse_obj(o, lambda o: o.clone().detach_().normal_())
```
Comment (kshitij12345) [inline test/test_modules.py:366]: Since the output of `LSTM` is (Tensor, (Tensor, Tensor))

File: test/test_modules.py [hunk 3]
```diff
@@ -447,7 +452,9 @@ def fn_to_gradcheck(*flat_input_and_params):
                 new_kwargs = {name: obj for (name, _), obj in zip(kwarg_tensors, kwarg_args)}
 
                 with freeze_rng_state():
-                    return m(*new_input_args, **new_kwargs, **other_kwargs)
+                    output = m(*new_input_args, **new_kwargs, **other_kwargs)
+                    output_flattened, _ = torch.utils._pytree.tree_flatten(output)
```
Comment (kshitij12345) [inline test/test_modules.py:456]: Since the output of `LSTM` is `(Tensor, (Tensor, Tensor))`

**Comment by kshitij12345** [comment]
@jbschlosser Gentle ping :)

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! Couple minor comments below then g2g :)

**Comment by kshitij12345** [comment]
Addressed the review. PTAL :)

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Comment by kshitij12345** [comment]
Gentle ping @jbschlosser  :)

**Comment by jbschlosser** [comment]
Hey @kshitij12345, I've been working through some internal breakage to get this landed. I got an internal team to land a fix last night that should help resolve some issues, so hopefully we're good to go now :)

**Comment by kshitij12345** [comment]
Ah. I thought maybe it just slipped the queue. Apologies for the noise :)
